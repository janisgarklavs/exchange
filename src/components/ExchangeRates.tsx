import React from 'react'

interface ExchangeRatesProps {
  rates: Record<string, number>
}

export const ExchangeRates: React.SFC<ExchangeRatesProps> = ({ rates }) => (
  <div style={{ padding: '12px 24px', textAlign: 'center' }}>
    {rates &&
      Object.keys(rates).map(rate => (
        <div key={rate} style={{ padding: '4px 0' }}>
          <b>{rate}</b> - {rates[rate]}
        </div>
      ))}
  </div>
)
