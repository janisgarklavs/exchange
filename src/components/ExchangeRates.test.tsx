import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import { ExchangeRates } from "./ExchangeRates";

describe("<ExchangeRates />", () => {
  test("it renders correctly", () => {
    const rates = { USD: 0.8 };
    const component = shallow(<ExchangeRates rates={rates} />);

    expect(toJson(component)).toMatchSnapshot();
  });
});
