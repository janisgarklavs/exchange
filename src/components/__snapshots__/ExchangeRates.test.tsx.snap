// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExchangeRates /> it renders correctly 1`] = `
<div
  style={
    Object {
      "padding": "12px 24px",
      "textAlign": "center",
    }
  }
>
  <div
    key="USD"
    style={
      Object {
        "padding": "4px 0",
      }
    }
  >
    <b>
      USD
    </b>
     - 
    0.8
  </div>
</div>
`;
